File: MVVM/Models/MealModel.cs
namespace HealthTrackerMVVM.Models
{
    using System.Text.Json.Serialization;

    public enum MealType
    {
        Breakfast,
        Lunch,
        Dinner,
        Snack,
    }

    public class MealModel
    {
        [JsonPropertyName("mealType")]
        [JsonConverter(typeof(JsonStringEnumConverter))]
        public MealType Type { get; set; }

        public int Calories { get; set; }

        public string DisplayName => $"{this.Type} - {this.Calories} kcal";
    }
}

File: API/src/main/java/com.healthtracker.tracker_api/model/MealModel.java
package com.healthtracker.tracker_api.model;

import lombok.Getter;

@Getter
public class MealModel {
    private final String username;
    private final String type;
    private final String calories;

    public enum MealType {
        Breakfast,
        Lunch,
        Dinner,
        Snack
    }

    public MealModel(String username, String type, String calories) {
        this.username = username;
        this.type = type;
        this.calories = calories;
    }
}
